articles/building-a-simple-cryptocurrency-blockchain/index.md

```diff
@@ -0,0 +1,187 @@
+
+
+Now that our blockchain is ready to work. We are missing something which is a core blockchain principle, the blockchain _integrity_. Let's see how to verify it and test our app.
+
+## Verifying Blockchain Integrity 
+
+The main characteristic of blockchain is that once a block has been added to the network, it can't be changed without invalidating the entire blockchain integrity.
+
+To perform this, we use digital security or cryptographic hash which ensures securing and validating of blockchain by producing a new hash every time any change is made in the block thus invalidating the blockchain.
+
+We'll loop over the entire blockchain to check whether any hash has been tampered with, taking into account the exception of the first block which is hardcoded.
+
+Besides, this method will verify if the hashes of each two consecutive blocks are pointing to one another. If the integrity of the blockchain has been compromised, it returns false; otherwise, in case no anomalies are encountered, it returns true.
+ We'll create this method inside inside `Blockchain` class
+```Node
+    checkValidity(){
+            for(let i = 1; i < this.block1chain.length; i++){
+                const currentBlock = this.block1chain[i];
+                const precedingBlock= this.blockchain[i-1];
+
+            if(currentBlock.hash !== currentBlock.computeHash()){
+                return false;
+            }
+            if(currentBlock.nextgHash !== nextBlock.hash)
+                return false;
+            }
+            return true;
+        }
+```js
+
+
+Now we can test our app and see the results:
+
+But before we dive into running the code, let's create a new instance of the `Blockchain` class and name it `thecoin`, and add some blocks in the blockchain using random values.  
+
+```Node
+  let thecoin = new Blockchain();
+    thecoin.addNewBlock(new BlockCrypto(1, "06/04/2021", {sender: "Rabin Yitzack", recipient: "Loyd Eve", quantity: 20}));
+    thecoin.addNewBlock(new BlockCrypto(2, "07/04/2021", {sender: "Anita Vyona", recipient: "Felix Mush", quantity: 349}) );
+    console.log(JSON.stringify(thecoin, null, 4));
+```js
+
+### Running our Blockchain
+
+Typing this command to our terminal `node thecoin.js` will result in this 
+
+![output](engineering-education/building-a-simple-cryptocurrency-blockchain/output.png)
+
+**NOTE:** >Before running the command ensure to navigate on the right path on your terminal.
+
+_hint:_ Using the command `pwd` to check the path.
+
+Here is our full source code
+
+## Conclusion
+
+Kudos! You are have just build a simple cryptocurrency app using Node.js, the step is closer to get you started building Pro apps using Node.js, or rather you can just add more features to our simple blockchain and thrill it to the market.
+Nevertheless, I hope that this tutorial has provided you with basic skill proficiency to get you going with the stimulating Node.js development.  
+
+Happy Coding!
+
+
```

Comment:
Add a `Further reading` section, that contains references or other links that would help the reader understand the topic better